src/LetsEncrypt.Azure/AzureKeyVaultExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Nate McMaster.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using McMaster.AspNetCore.LetsEncrypt;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extensions to enable Azure KeyVault connections with LetsEncrypt.
+    /// </summary>
+    public static class AzureKeyVaultExtensions
+    {
+        /// <summary>
+        /// Adds key vault certificate repository for LetsEncrypt.
+        /// </summary>
+        /// <param name="builder">A LetsEncrypt service builder.</param>
+        /// <param name="config">Configuration for KeyVault connections.</param>
+        /// <returns>The original LetsEncrypt service builder.</returns>
+        public static ILetsEncryptServiceBuilder AddKeyVaultCertificateRepository(this ILetsEncryptServiceBuilder builder, Action<AzureKeyVaultCertificateRepositoryOptions>? config = null)
```

Comment:
Can we name this `. PersistCertificatesToKeyVault()`? I'm going for a naming pattern here that is similar to the data protection APIs and already have `.PersistCertificatesToDirectory()`

Reply:
Since this both persists and retrieves from KeyVault, should it be separated to `.PersistCertifcatesToAzureKeyVault()` and `.AddAzureKeyVaultCertifcateStore()`?